app/ui/lib/Combobox.tsx

```diff
@@ -31,11 +31,17 @@ export type ComboboxBaseProps = {
   description?: React.ReactNode
   isDisabled?: boolean
   isLoading?: boolean
-  items: ComboboxItem[]
+  // without items, the combobox acts as a regular text input
+  items?: ComboboxItem[]
```

Comment:
Interesting, this hadn't occurred to me. I think you could still require it and just treat empty array the same as undefined. Because I don't see a place where we'd use this without there being a relevant list (which might happen to be empty).